mp/src/game/client/neo/ui/neo_scoreboard.cpp

```diff
@@ -738,10 +763,10 @@ void CNEOScoreBoard::GetPlayerScoreInfo(int playerIndex, KeyValues *kv)
 //-----------------------------------------------------------------------------
 // Purpose:
 //-----------------------------------------------------------------------------
-void CNEOScoreBoard::UpdatePlayerAvatar( int playerIndex, KeyValues *kv )
+void CNEOScoreBoard::UpdatePlayerAvatar( int playerIndex, KeyValues *kv, bool skipKey )
```

Comment:
Since we're already guarding for the nullptr case for the `kv` parameter, and it seems like the only place where we set the non-default `true` value for the newly introduced `skipKey` parameter also has the `kv` argument passed as nullptr:

```cpp
// At line 326 of mp/src/game/client/neo/ui/neo_scoreboard.cpp
UpdatePlayerAvatar(pPlayer->index, nullptr, true);
```

couldn't we simply rely on that in-band `kv` nullptr check, so we could get away without introducing the third `skipKey` parameter to the prototype, for simplicity? As both `!kv` and `!skipKey` are essentially doing the same work of avoiding the `kv` ptr indirection, unless I misunderstand the parameter's purpose.

Reply:
I was hesitant to modify the behavior of this function where it is called without the skipKey value set where kv may be missing, though in hindsight that is probably silly.